src/Simulation/QSharpFoundation/Convert/Convert.qs

```diff
@@ -164,7 +164,22 @@ namespace Microsoft.Quantum.Convert {
     /// the mappings of each Pauli operator in big-endian order
     /// `bits(Pn) ... bits(P0)`.
     function PauliArrayAsInt(paulis : Pauli[]) : Int {
-        body intrinsic;
+        let len = Length(paulis);
+        if len > 31 {
+            fail $"Cannot pack bits of Pauli array longer than 31 (got {len}).";
+        }
+
+        mutable result = 0;
+        for index in (len-1)..-1..0 {
+            let p = paulis[index];
```

Comment:
The index isn't actually used here, such that I would suggest replacing with a loop over `paulis` directly.

```suggestion
        for p in paulis[(len-1)..-1..0] {
```

Reply:
Changed. Thanks!